src/main/java/net/openhft/chronicle/bytes/internal/NativeBytesStore.java

```diff
@@ -705,7 +704,7 @@ public long readIncompleteLong(long offset) {
 
     @Override
     public boolean equals(Object obj) {
-        return HashCodeEqualsUtil.contentEquals(this, obj);
+        return obj instanceof BytesStore && BytesInternal.contentEqual(this, (BytesStore) obj);
     }
 
     public void setAddress(long address) {
```

Comment:
maybe add a @NoneNegitive

Reply:
There is a separate issue for this: https://github.com/OpenHFT/Chronicle-Bytes/issues/290

Reply:
for address? addresses can be -ve

Reply:
True. Addresses can be negative (e.g. for mapped files in Linux) but offsets and lengths are always non-negative.